server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraDumbBlobStoreCache.java

```diff
@@ -86,22 +86,25 @@ public CassandraDumbBlobStoreCache(Session session, CassandraCacheConfiguration
             insertStatement.bind()
                 .setString(ID, blobId.asString())
                 .setBytes(DATA, data)
-                .setInt(TTL, cacheConfiguration.getTtl())
+                .setInt(TTL, Long.valueOf(cacheConfiguration.getTtl().getSeconds()).intValue())
                 .setConsistencyLevel(ConsistencyLevel.ONE));
     }
 
     @Override
     public Publisher<Void> cache(BlobId blobId, InputStream inputStream) {
         Preconditions.checkNotNull(inputStream);
 
-        BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream, cacheConfiguration.getSizeThreshold() + 1);
-        return Mono.fromCallable(() -> isGreaterThanThreshold(bufferedInputStream))
-            .filter(bigStream -> !bigStream)
-            .flatMap(ignored ->
-                Mono.fromCallable(() -> IOUtils.toByteArray(bufferedInputStream))
-                    .flatMap(bytes -> Mono.from(save(blobId, ByteBuffer.wrap(bytes, 0, bytes.length)))))
-            .switchIfEmpty(Mono.empty())
-            .doFinally(any -> closeInputStreamQuite(inputStream));
+        BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream, cacheConfiguration.getByteThresholdSize() + 1);
+
+        return Mono.using(() -> isGreaterThanThreshold(bufferedInputStream),
```

Comment:
IHMO the ressource acquisition part of the using should be the creation of the bufferedInputStream here.